include/swift/AST/Types.h

```diff
@@ -2123,6 +2115,11 @@ class ParameterTypeFlags {
   uint16_t toRaw() const { return value.toRaw(); }
 };
 
+/// A type that indicates how parameter flags should be handled in an operation
+/// that requires the conversion into a type that doesn't support them, such as
+/// tuples. They must either be dropped, or be enforced to not be present.
+enum class ParameterFlagHandling { Drop, AssertEmpty };
```

Comment:
This is unsafe IHMO, be behavior should always be - assert if flags are present,  inside the solver `composeTuple` should only be called after appropriate checks have been performed. We really want to know about places where parameters with non-empty flags are converted into a tuple to figure out why the caller is not checking...

Reply:
Talked with Pavel offline, we're going to keep this as-is, but with `Drop` renamed to `IgnoreNonEmpty`. It turns out that most callers really do want to drop flags, and making this more convoluted runs the risk of callers just building the tuple themselves without explicitly calling out the fact they are dropping flags.